### ISSUE TITLE
How to accelerate the running speed

### ISSUE THREAD

**Issue by ZrW00** [steveli/pytorch-sqrtm#2]
I try to calculate sqrtm by this method, but soon I found it so slow to run, How can I accelerate the running speed?

**Comment by h-gokul** [comment]
I dont think we can do much here. Most of the latency s behould due to converting your data from GPU to cpu back and forth and not necessarily suboptimal code.  If you know a better implementation please let meknow.

**Comment by steveli** [comment]
I find discussions in this thread useful: https://github.com/pytorch/pytorch/issues/25481
